ArmPkg/Universal/Smbios/SmbiosMiscDxe/Type03/MiscChassisManufacturerFunction.c

```diff
@@ -98,71 +93,72 @@ SMBIOS_MISC_TABLE_FUNCTION (MiscChassisManufacturer) {
 
   TokenToGet   = STRING_TOKEN (STR_MISC_CHASSIS_MANUFACTURER);
   Manufacturer = HiiGetPackageString (&gEfiCallerIdGuid, TokenToGet, NULL);
-  ManuStrLen   = StrLen (Manufacturer);
+  ManuStrSize  = StrSize (Manufacturer);
 
   TokenToGet = STRING_TOKEN (STR_MISC_CHASSIS_VERSION);
   Version    = HiiGetPackageString (&gEfiCallerIdGuid, TokenToGet, NULL);
-  VerStrLen  = StrLen (Version);
+  VerStrSize = StrSize (Version);
 
-  TokenToGet      = STRING_TOKEN (STR_MISC_CHASSIS_SERIAL_NUMBER);
-  SerialNumber    = HiiGetPackageString (&gEfiCallerIdGuid, TokenToGet, NULL);
-  SerialNumStrLen = StrLen (SerialNumber);
+  TokenToGet       = STRING_TOKEN (STR_MISC_CHASSIS_SERIAL_NUMBER);
+  SerialNumber     = HiiGetPackageString (&gEfiCallerIdGuid, TokenToGet, NULL);
+  SerialNumStrSize = StrSize (SerialNumber);
 
-  TokenToGet      = STRING_TOKEN (STR_MISC_CHASSIS_ASSET_TAG);
-  AssertTag       = HiiGetPackageString (&gEfiCallerIdGuid, TokenToGet, NULL);
-  AssertTagStrLen = StrLen (AssertTag);
+  TokenToGet       = STRING_TOKEN (STR_MISC_CHASSIS_ASSET_TAG);
+  AssertTag        = HiiGetPackageString (&gEfiCallerIdGuid, TokenToGet, NULL);
+  AssertTagStrSize = StrSize (AssertTag);
 
   TokenToGet       = STRING_TOKEN (STR_MISC_CHASSIS_SKU_NUMBER);
   ChassisSkuNumber = HiiGetPackageString (&gEfiCallerIdGuid, TokenToGet, NULL);
-  ChaNumStrLen     = StrLen (ChassisSkuNumber);
+  ChaNumStrSize    = StrSize (ChassisSkuNumber);
 
-  ContainedElementCount = InputData->ContainedElementCount;
-  ExtendLength          = ContainedElementCount * sizeof (CONTAINED_ELEMENT);
+  BaseSize   = sizeof (SMBIOS_TABLE_TYPE3) - sizeof (CONTAINED_ELEMENT);

```

Comment:
This is risky. It might work in this particular case, but in general, you should take padding into account, which might precede the `ContaintedElements[1]` member. Better to use `OFFSET_OF(SMBIOS_TABLE_TYPE3, ContainedElemenst)` here - this is also guaranteed to work as expected regardless of whether `ContainedElements` is a fake flex array `[1]` or a true one `[]`.